ironfish-zkp/src/primitives/proof_generation_key.rs

```diff
@@ -0,0 +1,200 @@
+use group::GroupEncoding;
+use jubjub::{Fr, SubgroupPoint};
+use std::error::Error;
+use std::fmt;
+use std::ops::Deref;
+use zcash_primitives::sapling::ProofGenerationKey as ZcashProofGenerationKey;
+
+#[derive(Debug)]
+pub enum ProofGenerationKeyError {
+    InvalidAuthorizingKey,
+    InvalidNullifierDerivingKey,
+    HexConversionError,
+}
+
+impl fmt::Display for ProofGenerationKeyError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ProofGenerationKeyError::InvalidAuthorizingKey => write!(f, "Invalid authorizing key"),
+            ProofGenerationKeyError::InvalidNullifierDerivingKey => {
+                write!(f, "Invalid nullifier deriving key")
+            }
+            ProofGenerationKeyError::HexConversionError => write!(f, "Hex conversion error"),
+        }
+    }
+}
+
+impl Error for ProofGenerationKeyError {}
+
+#[derive(Clone)]
+pub struct ProofGenerationKey(ZcashProofGenerationKey);
+
+impl ProofGenerationKey {
+    pub fn new(ak: SubgroupPoint, nsk: Fr) -> Self {
+        ProofGenerationKey(ZcashProofGenerationKey { ak, nsk })
+    }
+
+    pub fn to_bytes(&self) -> [u8; 64] {
+        let mut proof_generation_key_bytes: [u8; 64] = [0; 64];
+        proof_generation_key_bytes[0..32].copy_from_slice(&self.0.ak.to_bytes());
+        proof_generation_key_bytes[32..].copy_from_slice(&self.0.nsk.to_bytes());
+        proof_generation_key_bytes
+    }
+
+    pub fn from_bytes(
+        proof_generation_key_bytes: [u8; 64],
```

Comment:
Nit: I would change this from `[u8; 64]` to `&[u8; 64]` or (even better, but requires a few extra code) `&[u8]`